tests/robustness/traffic/key_window.go

```diff
@@ -0,0 +1,62 @@
+package traffic
+
+import (
+	"fmt"
+	"math/rand"
+	"sync"
+)
+
+type KeyWindow struct {
+	sync.Mutex
+
+	Counter    int
+	WindowSize int
+	Keys       []int
+	keyPrefix  string
+}
+
+func NewKeyWindow(size int, keyPrefix string) *KeyWindow {
+	k := &KeyWindow{
+		Counter:    0,
+		WindowSize: size,
+		Keys:       make([]int, size),
+		keyPrefix:  keyPrefix,
+	}
+
+	for k.Counter < size {
+		k.Keys[k.Counter] = k.Counter
+		k.Counter++
+	}
+
+	return k
+}
+
+func (k *KeyWindow) GetKey() string {
+	k.Lock()
+	defer k.Unlock()
+
+	if k.useWindow() {
+		// This will use the keys in the window
+		randKey := rand.Intn(k.WindowSize)
+		key := k.Keys[randKey]
+		k.Keys[randKey] = k.Counter
+		k.Counter++
+
+		return fmt.Sprintf("%s%d", k.keyPrefix, key)
+	}
+
+	// This will use the default key0 to key9
+	randKey := rand.Intn(k.WindowSize)
+	key := randKey
+
+	return fmt.Sprintf("%s%d", k.keyPrefix, key)
+}
+
+func (k *KeyWindow) useWindow() bool {
+	randVal := rand.Intn(100)
```

Comment:
This function can be simplified to be something like
```suggestion
return !rand.Intn(100) < 10
```